src/System.Numerics.Matrices/src/System/Numerics/IMatrix.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace System.Numerics.Matrices
+{
+    interface IMatrix
+    {
+        double this[int col, int row] { get; set; }
```

Comment:
In my experience (mostly in the world of graphics) `double` is the most useful (though I could easily argue for `float` as well). Since there is no `<T> where T is intrinsic` or equivalent, I cannot  make these generic. Therefore `double` was the closest thing as it solves the widest range of issues.

 * supports positive and negative values
 * supports a very wide range of values with decent precision (thanks to 64-bitness)
 * supports infinity
 * supports NaN
 * has built in CPU and GPU support
 * etc